test/Makefile

```diff
@@ -28,6 +28,12 @@
 #   EXECUTE_ARGS:        parameters to add to the execution of the test
 #                        default: (none)
 #
+#   COMPILED_IMPORTS:    list of modules files that are imported by the main source file that
+#                        should be included in compilation; this differs from the EXTRA_SOURCES
+#                        variable in that these files could be compiled by either explicitly
+#                        passing them to the compiler or by using the "-i" option.
+#                        default: (none)
```

Comment:
Why can't this be united with `EXTRA_SOURCES`? If this option is superior, `EXTRA_SOURCES` could be removed?

Reply:
COMPILE_IMPORTS is different from EXTRA_SOURCES in that EXTRA_SOURCES are source files that cannot be added via the `-i` option.  This applies to documentation files for example.  COMPILE_IMPORTs on the other hand are modules that can be added via `-i`.  The test has been modified to perform the compilation using both methods (using `-i` and explicitly giving modules on the commnand line), but only if the modules are in the COMPILE_IMPORTS variable.

Reply:
> The test has been modified to perform the compilation using both methods (using -i and explicitly giving modules on the commnand line), but only if the modules are in the COMPILE_IMPORTS variable.

Yeah your motivation - that's what I was asking for (would be great to mention this in the overview).
I'm not sure it's even necessary to do the compilation twice (it just slows down the testsuite). After all, `-i` is just a frontend feature and you already have quite a few tests for it.